src/app/dataset-block/metadata-block/metadata-block.component.ts

```diff
@@ -20,10 +27,42 @@ export class MetadataBlockComponent
     extends BaseProcessingComponent
     implements OnInit
 {
+    constructor(
+        private datasetService: DatasetService,
+        private router: Router,
+        navigationService: NavigationService,
+        modalService: ModalService,
+        activatedRoute: ActivatedRoute,
+    ) {
+        super(navigationService, modalService, activatedRoute);
+    }
     public datasetInfo: DatasetInfo;
     public datasetViewType = DatasetViewTypeEnum.History;
+    public blockHash: string;
 
     ngOnInit(): void {
         this.datasetInfo = this.getDatasetInfoFromUrl();
+        this.trackSubscriptions(
+            this.activatedRoute.params
+                .pipe(pluck(ProjectLinks.URL_PARAM_BLOCK_HASH))
+                .subscribe((hash: string) => {
+                    this.blockHash = hash;
+                }),
+            this.router.events
+                .pipe(
+                    filter((event) => event instanceof NavigationEnd),
+                    tap(() => {
+                        this.getMetadataBlock();
```

Comment:
Shouldn't this subscription be tracked too?

Reply:
Yes, added this invoke to track subscription